### DISCUSSION TITLE
FileAttachement[parquet]: "Failed to bundle using Rollup"

### DISCUSSION THREAD

**Discussion by e-roux** [observablehq/framework#1302] [Q&A]
Hello 👋,

using observable framework 1.7.0 with bun/yarn and doing something relatively simple in a md file:

  ```js
  const titanic = FileAttachment("data/titanic.parquet").parquet();
  ``` 
  
  ```js
  view(titanic.schema)
  ```

Log from the terminal:

```shell-output
GET /great_stuff
npm:@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-mvp.worker.js → src/.observablehq/cache/_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-mvp.worker.js
npm:@duckdb/duckdb-wasm@1.28.0/dist/duckdb-eh.wasm → src/.observablehq/cache/_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-eh.wasm
npm:@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-eh.worker.js → src/.observablehq/cache/_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-eh.worker.js
npm:parquet-wasm@0.6.0/esm/parquet_wasm_bg.wasm → src/.observablehq/cache/_npm/parquet-wasm@0.6.0/esm/parquet_wasm_bg.wasm
socket close /_observablehq
GET /_npm/apache-arrow@16.0.0/_esm.js
GET /_npm/parquet-wasm@0.6.0/_esm.js
GET /_npm/@duckdb/duckdb-wasm@1.28.0/_esm.js
GET /_npm/htl@0.3.1/_esm.js
GET /_npm/tslib@2.6.2/_esm.js
GET /_npm/flatbuffers@23.5.26/_esm.js
socket open /_observablehq
↑ {
  type: 'hello',
  path: '/great_stuff',
  hash: 'b6fbecbc6f4424f7b3c602524d5e61bee861713507d5ba34a0c7e1dee5709110'
}
GET /_npm/apache-arrow@15.0.2/_esm.js
npm:apache-arrow@15.0.2/+esm → src/.observablehq/cache/_npm/apache-arrow@15.0.2/_esm.js
GET /_npm/parquet-wasm@0.5.0/_esm.js
npm:parquet-wasm@0.5.0/+esm → src/.observablehq/cache/_npm/parquet-wasm@0.5.0/_esm.js
GET /_file/data/kaggle.parquet?sha=96ac672eeaa54bb72dd0cc6e82fb9a791c80978914a1df9ed4cd2fd899eafcc3
GET /_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-eh.worker.js
GET /_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-eh.wasm
```

And in the browser, an error:
```shell-output
RuntimeError: Failed to bundle using Rollup v2.79.1: failed to resolve an internal import. If you believe this to be an issue with jsDelivr, and not with the package itself, please open an issue at https://github.com/jsdelivr/jsdelivr
```

I don't really understand with two versions of parquet are being loaded. Any idea?

**Comment by mbostock** [comment]
First, I would first try clearing your npm cache; you probably have the older version of parquet-wasm cached here.

```sh
rm -rvf src/.observablehq/cache/_npm
```

Second, you’ll want to use `display` instead of `view`; `view` is for [declaring inputs](https://observablehq.com/framework/reactivity#inputs).

Your logs output suggest that you’re using DuckDB, which isn’t evident from your code, so perhaps there’s something else that’s going on here? This code works for me to load a Parquet file:

````md
```js
const gaia = FileAttachment("lib/gaia-sample.parquet").parquet();
```
```js
display(gaia);
```
````

If you want to use DuckDB, this also worked for me using Framework’s SQL cells:

````md
---
sql:
  gaia: lib/gaia-sample.parquet
---

```sql
SELECT * FROM gaia LIMIT 10
```
````

Of course you can also use `DuckDBClient` directly like so:

````md
```js
const gaia = FileAttachment("lib/gaia-sample.parquet").parquet();
```
```js
const db = DuckDBClient.of({gaia});
```
```js
db.sql`SELECT * FROM gaia LIMIT 10`
```
````

**Comment by e-roux** [comment]
Thank you for the quick answer and sorry for the noise. I indeed cleared the node cache several times before opening an issue, but always got the error. As you noticed it, I'm experimenting with various input and performed an update 1.6.0->1.7.0. I wonder if the browser's cache is faulty as I do not have the issue when opening a private window. I'm on Safari/MacOS.

The same uri called from two windows:

```shell-session
socket open /_observablehq
↑ {
  type: 'hello',
  path: '/great_stuff',
  hash: '170f0f8eabe4b13d30bc846d8b26d4bb97b1722c35c4f281176dd1db6cb16be6'
}
GET /_npm/apache-arrow@15.0.2/_esm.js
GET /_npm/parquet-wasm@0.5.0/_esm.js
GET /_file/data/kaggle.parquet?sha=96ac672eeaa54bb72dd0cc6e82fb9a791c80978914a1df9ed4cd2fd899eafcc3
GET /great_stuff
socket open /_observablehq
↑ {
  type: 'hello',
  path: '/great_stuff',
  hash: '170f0f8eabe4b13d30bc846d8b26d4bb97b1722c35c4f281176dd1db6cb16be6'
}
GET /_file/data/kaggle.parquet?sha=96ac672eeaa54bb72dd0cc6e82fb9a791c80978914a1df9ed4cd2fd899eafcc3
GET /_npm/parquet-wasm@0.6.0/esm/parquet_wasm_bg.wasm
GET /_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-browser-eh.worker.js
GET /_npm/@duckdb/duckdb-wasm@1.28.0/dist/duckdb-eh.wasm
```

**Reply by mbostock** [reply]
Ah, yes, Safari has problems with caching sometimes. Try shift-clicking the reload button? You could also trying [clearing website data](https://support.apple.com/guide/safari/manage-cookies-sfri11471/17.0/mac/14.0) (which I think clears the cache in Safari — this isn’t about cookies).

**Reply by mbostock** [reply]
Also did you clear the `_node` cache or the `_npm` cache? (The `_node` cache is for imports from `node_modules` managed by a package manager, whereas the `_npm` cache is for imports from npm via jsDelivr.)

**Comment by e-roux** [comment]
I removed the whole `src/.observablehq/cache` directory, and rebuild everything multiple times. Now I closed all the observable windows and it's fine. Clearing the cache in Safari would have solved the problem. I'll test with the next release because I had the same problem when upgrading to 1.6.0
